primitives/polkadot-core/src/lib.rs

```diff
@@ -336,12 +340,12 @@ where
 
 	fn pre_dispatch(
 		self,
-		who: &Self::AccountId,
-		call: &Self::Call,
-		info: &DispatchInfoOf<Self::Call>,
-		len: usize,
+		_who: &Self::AccountId,
+		_call: &Self::Call,
+		_info: &DispatchInfoOf<Self::Call>,
+		_len: usize,
 	) -> Result<Self::Pre, TransactionValidityError> {
-		Ok(self.validate(who, call, info, len).map(|_| ())?)
```

Comment:
Do we forgo validation completely? Is this because we can't validate encoded call?

Reply:
Yeah - this function is only called within runtime and we're using `SignedExtensions` just to get the encoded data (so we don't need any validation at all). More over, the default `SignedExtension::validate` actually does nothing: https://github.com/paritytech/substrate/blob/master/primitives/runtime/src/traits.rs#L862-L879